pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -33,11 +33,14 @@ import (
 // for managing gRPC communications with the device plugin and caching
 // device states reported by the device plugin.
 type endpoint interface {
-	run()
-	stop()
-	allocate(devs []string) (*pluginapi.AllocateResponse, error)
-	getDevices() []pluginapi.Device
-	callback(resourceName string, added, updated, deleted []pluginapi.Device)
+	Run()
+	Stop() error
+
+	Allocate(devs []string) (*pluginapi.AllocateResponse, error)
+	ResourceName() string
+
+	Store() deviceStore
+	SetStore(deviceStore)
```

Comment:
Why we change these functions to be public?

Reply:
Answered above :)